src/components/v2/deployment-history/ea/EADeploymentHistory.component.tsx

```diff
@@ -320,27 +334,86 @@ function ExternalAppDeploymentHistory({ appId }: { appId: string }) {
                                         </Tippy>
                                     </div>
                                 )
-                            })
-                        }
+                            })}
+                        </div>
                     </div>
+                    {selectedDeploymentHistoryIndex !== 0 && (
+                        <Tippy className="default-tt" arrow={false} content={'Re-deploy this version'}>
+                            <button
+                                className="flex cta deploy-button"
+                                onClick={() => setShowRollbackConfirmation(true)}
+                            >
+                                <DeployButton className="deploy-button-icon" />
+                                <span className="ml-4">Deploy</span>
+                            </button>
+                        </Tippy>
+                    )}
                 </div>
             </div>
-        </div>
+        )
     }
 
+    async function handleDeployClick() {
+        try {
+            const selectedDeploymentHistory = deploymentHistoryArr[selectedDeploymentHistoryIndex]
+            const requestPayload: RollbackReleaseRequest = {
+                hAppId: appId,
+                version: selectedDeploymentHistory.version,
+            }
+
+            if (installedAppInfo) {
+                requestPayload.installedAppId = installedAppInfo.installedAppId
+                requestPayload.installedAppVersionId = installedAppInfo.installedAppVersionId
+            }
+
+
+            setShowRollbackConfirmation(false)
+            history.push(`${url.split('/').slice(0, -1).join('/')}/${URLS.APP_DETAILS}?refetchData=true`)
+
+            await rollbackApplicationDeployment(requestPayload, !installedAppInfo)
+            toast.success('Deployment initiated')
+        } catch (err) {
+            setShowRollbackConfirmation(false)
+            showError(err)
+        }
+    }
+
+    function RollbackConfirmationDialog() {
+        return (
+            <ConfirmationDialog className="rollback-confirmation-dialog">
+                <ConfirmationDialog.Body title={`Rollback "${appName}"`}>
+                    <p className="fs-13 cn-7 lh-1-54">Are you sure you want to deploy a previous version?</p>
```

Comment:
IMO we should show some more info to user (for eg : version)

Reply:
Checked & confirmed. For now, we'll be sticking to the current info message.